recipe/meta.yaml

```diff
@@ -29,33 +29,42 @@ outputs:
       run_exports:
         - {{ pin_subpackage('openmpi', min_pin='x.x.x', max_pin='x') }}
       ignore_run_exports:
-        - ucx  # [enable_cuda]
+        - ucx  # [enable_ucx]
       script_env:
-        - CUDA_HOME     # [enable_cuda]
+        - CUDA_HOME  # [enable_cuda]
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - {{ compiler('fortran') }}
-        - autoconf  # [unix]
-        - automake  # [unix]
+        #- autoconf  # [unix]
+        #- automake  # [unix]
         - libtool   # [unix]
         - make  # [unix]
-        - perl 5.26.2
       host:
+        #- openpmix
+        #- prrte
+        - libhwloc
+        - libevent
         - zlib
-        - ucx                 # [enable_cuda]
-        - ucx-proc =*=gpu     # [enable_cuda]
+        - ucx  # [enable_ucx]
+        - cuda-version {{ cuda_version }}  # [enable_cuda]
       run:
-        - zlib
         - mpi 1.0 openmpi
+        #- openpmix
+        #- prrte
+        - libhwloc
+        - libevent
+        - zlib
       run_constrained:
-        - cudatoolkit  >= {{ cudatoolkit }}         # [enable_cuda]
-        - {{ pin_compatible("ucx", max_pin="x") }}  # [enable_cuda]
+        - {{ pin_compatible("ucx", max_pin="x.x") }}  # [enable_ucx]
+        # Open MPI only uses CUDA Driver APIs, set the minimal driver version
+        - __cuda  >= {{ cuda_version }}  # [enable_cuda]
```

Comment:
@leofang Can you teach me what this double leading underscore means?

Reply:
This gives an overview: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-virtual.html

Reply:
Sorry I was out over the long weekend. As Jaimed pointed out (thanks!) there are a few pre-defined virtual packages in conda to help constrain at the system level. `__cuda` is used to detect the installed CUDA user-mode driver (`libcuda.so`) version. Here, we don't really care its version, other than ensuring user's driver is equal to or higher than the one used at build time.